src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -212,13 +222,16 @@ private void OnProjectsChanged(ConnectionInformation connection, IEnumerable<Son
 
         private SonarQubeOrganization AskUserToSelectOrganizationOnUIThread(IEnumerable<SonarQubeOrganization> organizations)
         {
-            return Application.Current.Dispatcher.Invoke(() =>
+            SonarQubeOrganization organization = null;
+
+            threadHandling.RunOnUIThread2(() =>
             {
                 var organizationDialog = new OrganizationSelectionWindow(organizations) { Owner = Application.Current.MainWindow };
                 var hasUserClickedOk = organizationDialog.ShowDialog().GetValueOrDefault();
 
-                return hasUserClickedOk ? organizationDialog.Organization : null;
+                organization = hasUserClickedOk? organizationDialog.Organization : null;
             });
+            return organization;
```

Comment:
I manually debugged the product code and confirmed it is switching to/from the UI thread as expected.